File: solutions/507.perfect-number/perfect-number.js
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function(num) {
    if(num <= 1) return false;
    let sum = 1;
    for(let i = 2; i <= Math.floor(Math.sqrt(num)); i++){
        if(num % i === 0){
            sum += i + num / i;
        }
    }
    if(sum === num) return true;
    return false;
};

File: solutions/507.perfect-number/perfect-number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        if(num <=0) return false;
        int sum =0;
        for(int i = 1; i <= Math.sqrt(num); i++){
            if(num % i == 0){
                if(i != num){
                    sum += i;
                }
                if( num/i != num){
                    sum += num / i;
                }
            }
        }
        if(sum == num){
            return true;
        }
        return false;
    }
}